central/src/gitlab.rs

```diff
@@ -0,0 +1,105 @@
+use beam_lib::reqwest::Url;
+use clap::Parser;
+use serde::Deserialize;
+use serde_json::json;
+use shared::{GitLabProjectAccessTokenConfig, SecretResult};
+
+#[derive(Parser)]
+struct GitLabApiConfig {
+    /// The base URL for API calls including the trailing slash, e.g. "https://gitlab.com/"
```

Comment:
The info "including the trailing slash" is uncommon and hints to me that you might be building the URL by hand (see comment below)